File: romantoint.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char ,int> romanMap;
        romanMap['I'] = 1;
        romanMap['X'] = 10;
        romanMap['V'] = 5;
        romanMap['L'] = 50;
        romanMap['C'] = 100;
        romanMap['M'] = 1000;
        romanMap['D'] = 500;
        int ans = 0 ;
        for(int i{0} ; i <= s.length() ; i++)
        {
            if(romanMap[s[i]] < romanMap[s[i+1]])
            {
                ans -= romanMap[s[i]];
            }
            else
            {
                ans += romanMap[s[i]];
            }
        }

        return ans;

        
    }
};

File: romantoint.java
public class RomanToInt {
    public static int romantoInt(String s)
    {
        int ans = 0 ;
        int num = 0 ;
        int prev = 0 ;
        for (int i = 0 ; i <= s.length() ; i++ )
        {
            switch(s.charAt(i))
            {
                case 'I' : num = 1 ; break ;
                case 'V' : num = 5 ; break ;
                case 'X' : num = 10 ; break ;
                case 'L' : num = 50 ; break ;
                case 'C' : num = 100 ; break ;
                case 'D' : num = 500 ; break ;
                case 'M' : num = 1000 ; break ;
            }
            if(num < prev)
            {
                ans -= num ;
            }
            else{
                ans += num ;
            }
            prev = num;
        }
        return ans;
    }

    public static void main(String s[])
    {
       System.out.println(romantoInt("XIV"));
    }
    
}
